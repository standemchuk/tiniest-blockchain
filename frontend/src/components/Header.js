import React from 'react'
import { Link } from 'react-router-dom'
import AppBar from '@material-ui/core/AppBar'
import Button from '@material-ui/core/Button'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

const styles = () => ({
  name: {
    flexGrow: 1
  }
})

export const Header = (props) => (
  <AppBar color='default' position='static'>
    <Toolbar>
      <Typography
        style={{ textDecoration: 'none' }}
        component={Link}
        to={'/'}
        variant='title'
        color='inherit'
        className={props.classes.name}
      >
        Tiniest Blockchain
      </Typography>
      <Button component={Link} to='/' color='inherit'>Account balance</Button>
      <Button component={Link} to='/transaction-history' color='inherit'>
        Transaction history
      </Button>
      <Button component={Link} to={'/create-transaction'} color='secondary'>Create transaction</Button>
    </Toolbar>
  </AppBar>
)

export default withStyles(styles)(Header)
