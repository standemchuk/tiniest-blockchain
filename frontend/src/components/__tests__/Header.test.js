import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { Header } from '../Header'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Header classes={{ name: 'name' }} />
      </MemoryRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
