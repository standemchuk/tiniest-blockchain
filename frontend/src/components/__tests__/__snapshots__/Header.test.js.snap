// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="MuiPaper-root-9 MuiPaper-elevation4-15 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-6"
>
  <div
    className="MuiToolbar-root-36 MuiToolbar-regular-38 MuiToolbar-gutters-37"
  >
    <a
      className="MuiTypography-root-40 MuiTypography-title-46 MuiTypography-colorInherit-59 name"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      Tiniest Blockchain
    </a>
    <a
      className="MuiButtonBase-root-88 MuiButton-root-64 MuiButton-text-66 MuiButton-flat-69 MuiButton-colorInherit-83"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-65"
      >
        Account balance
      </span>
      <span
        className="MuiTouchRipple-root-91"
      />
    </a>
    <a
      className="MuiButtonBase-root-88 MuiButton-root-64 MuiButton-text-66 MuiButton-flat-69 MuiButton-colorInherit-83"
      href="/transaction-history"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-65"
      >
        Transaction history
      </span>
      <span
        className="MuiTouchRipple-root-91"
      />
    </a>
    <a
      className="MuiButtonBase-root-88 MuiButton-root-64 MuiButton-text-66 MuiButton-textSecondary-68 MuiButton-flat-69 MuiButton-flatSecondary-71"
      href="/create-transaction"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiButton-label-65"
      >
        Create transaction
      </span>
      <span
        className="MuiTouchRipple-root-91"
      />
    </a>
  </div>
</header>
`;
